registration/include/pcl/registration/impl/gicp.hpp

```diff
@@ -177,9 +177,9 @@ GeneralizedIterativeClosestPoint<PointSource, PointTarget>::computeRDerivative(
   dR_dPsi(1, 2) = sphi * spsi + cphi * cpsi * stheta;
   dR_dPsi(2, 2) = 0.;
 
-  g[3] = matricesInnerProd(dR_dPhi, R);
-  g[4] = matricesInnerProd(dR_dTheta, R);
-  g[5] = matricesInnerProd(dR_dPsi, R);
+  g[3] = matricesInnerProd(dR_dPhi, dCost_dR_T);
+  g[4] = matricesInnerProd(dR_dTheta, dCost_dR_T);
+  g[5] = matricesInnerProd(dR_dPsi, dCost_dR_T);
```

Comment:
It seems `dR_dPsi` and `dCost_dR_T` should be swapped?

Reply:
With the formula `tr(A^TB) = tr(AB^T) = tr(B^TA) = tr(BA^T)`, the original implementation should be correct.